runtime/src/main/java/org/corfudb/runtime/collections/StreamListenerResumeOrDefault.java

```diff
@@ -0,0 +1,51 @@
+package org.corfudb.runtime.collections;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.exceptions.StreamingException;
+
+import java.util.List;
+
+/**
+ * This is an abstract stream listener callback implementation for clients
+ * interested in automatic re-subscription upon any error.
+ * The re-subscription policy of this implementation is an attempt to
+ * optimize and resume streaming from the last processed entry. If and only if
+ * we can't resume from the last processed entry, we subscribe from the latest
+ * position in the log (default), which can incur in data loss, as the latest
+ * position in the log might have progressed beyond several updates from
+ * the last processed entry.
+ *
+ * Please note if data loss is a concern DO NOT USE this listener implementation
+ * and refer to StreamListenerResumeOrFullSync
+ *
+ * Created by annym on 06-22-2021
+ */
+@Slf4j
+public abstract class StreamListenerResumeOrDefault extends StreamListenerResumePolicy {
```

Comment:
One general concern I have is I see a few implementations of StreamListener extending other classes, which will break since java does not support multiple inheritance but that is something we just need to help folks with.

Reply:
I see, this is valid. But I'm not sure how we could with this on our side. In any case, these 'patterns' are optional and available just to ease their job and for all boilerplate code. So they can still decide not to use these patterns directly.